alica_engine/include/engine/RuleBook.h

```diff
@@ -26,14 +26,17 @@ class AlicaEngine;
 class TeamObserver;
 class TeamManager;
 class IAlicaWorldModel;
+class ConfigChangeListener;
+class PlanRepository;
 
 /**
  * Defines the operational semantics of the used ALICA dialect.
  */
 class RuleBook
 {
 public:
-    RuleBook(AlicaEngine* ae, PlanBase* pb);
+    RuleBook(ConfigChangeListener& configChangeListener, Logger& log, SyncModule& synchModule, TeamObserver& teamObserver, TeamManager& teamManager,
```

Comment:
TeamManager stored as const, but not passed to the constructor as const. Is that intentional?